### ISSUE TITLE
Question about training loss

### ISSUE THREAD

**Issue by mxjmtxrm** [MeetKai/functionary#207]
Hi, great work!
I want to do llama sft with packing dataset. How to deal with the loss in the forward to ensure loss a+loss b=loss(pack(a,b))? Is there a monkey patch?

**Comment by khai-meetkai** [comment]
Hi @mxjmtxrm yes, we need to [monkey-patch](https://github.com/MeetKai/functionary/blob/main/functionary/train/packing/monkey_patch_packing.py#L126) and also pack the inputs as follows: https://github.com/MeetKai/functionary/tree/main/functionary/train/packing#convert-to-packed-dataset

**Comment by mxjmtxrm** [comment]
Thanks for your reply @khai-meetkai.
The monkey patch process the packed data. The loss didn't be updated right? [modeling_llama.py ](https://github.com/huggingface/transformers/blob/348e2294ac1a3cc744a93f04ab1f886439ee90d3/src/transformers/models/llama/modeling_llama.py#L1398)

**Comment by khai-meetkai** [comment]
@mxjmtxrm, yes the loss wasn't changed
